alephclient/crawldir.py

```diff
@@ -77,17 +92,22 @@ def scandir(self, path: PathLike, id: str, parent_id: str):
                 else:
                     self.queue.put((child, id))
 
-    def get_foreign_id(self, path: PathLike) -> Optional[str]:
+    def get_foreign_id(self, path: Path) -> Optional[str]:
         if path == self.root:
             if path.is_dir():
                 return None
             return path.name
-        path = PurePath(path)
-        if path.is_relative_to(self.root):
+
+        path = PurePath(path)  # type: ignore
```

Comment:
If you use `Path`, you won't need to use the ignore statement I think

Reply:
It needs to be converted to a PurePath. If you don't the system will fail, when directories are passed into the function the is_relative call fails

Reply:
Since this function expects path to be a instance of `Path`, we should make sure the calling function does that conversion if it's necessary.

Basically changing `self.get_foreign_id(path))` to `self.get_foreign_id(Path(path)))` in a couple of places in this file should take care of that.